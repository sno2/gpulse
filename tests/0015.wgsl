// array<f32,8> and array<i32,8> are different types:
// different element types
var<private> a: array<f32,8>;
var<private> b: array<i32,8>;
var<private> c: array<i32,8u>;  // array<i32,8> and array<i32,8u> are the same type

const width = 8;
const height = 8;

// array<i32,8>, array<i32,8u>, and array<i32,width> are the same type.
// Their element counts evaluate to 8.
var<private> d: array<i32,width>;

// array<i32,height> and array<i32,width> are the same type.
var<private> e: array<i32,width>;
var<private> f: array<i32,height>;
