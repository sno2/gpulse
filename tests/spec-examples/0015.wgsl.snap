ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 94, .end = 95 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 97, .end = 102 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 103, .end = 106 } }, ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 107, .end = 108 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } }, .value = null } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 124, .end = 125 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 127, .end = 132 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 133, .end = 136 } }, ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 137, .end = 138 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } } }, .value = null } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 154, .end = 155 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 157, .end = 162 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 163, .end = 166 } }, ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 167, .end = 169 }, .kind = ast.NumberLiteral.NumberKind.u32 } } } } }, .value = null } }
ast.Node{ .const_decl = ast.ConstDecl{ .name = ast.Span{ .start = 232, .end = 237 }, .typ = null, .value = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 240, .end = 241 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } }
ast.Node{ .const_decl = ast.ConstDecl{ .name = ast.Span{ .start = 249, .end = 255 }, .typ = null, .value = ast.Node{ .number_literal = ast.NumberLiteral{ .span = ast.Span{ .start = 258, .end = 259 }, .kind = ast.NumberLiteral.NumberKind.abstract_int } } } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 386, .end = 387 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 389, .end = 394 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 395, .end = 398 } }, ast.Node{ .identifier = ast.Span{ .start = 399, .end = 404 } } } } }, .value = null } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 482, .end = 483 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 485, .end = 490 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 491, .end = 494 } }, ast.Node{ .identifier = ast.Span{ .start = 495, .end = 500 } } } } }, .value = null } }
ast.Node{ .var_decl = ast.VarDecl{ .name = ast.Span{ .start = 516, .end = 517 }, .access_mode = null, .addr_space = ast.AddrSpace.private, .typ = ast.Node{ .template = ast.Template{ .name = ast.Span{ .start = 519, .end = 524 }, .args = { ast.Node{ .identifier = ast.Span{ .start = 525, .end = 528 } }, ast.Node{ .identifier = ast.Span{ .start = 529, .end = 535 } } } } }, .value = null } }
